lnwallet/interface_test.go

```diff
@@ -2376,6 +2372,79 @@ func testChangeOutputSpendConfirmation(r *rpctest.Harness,
 	}
 }
 
+// testSpendUnconfirmed ensures that when can spend unconfirmed outputs.
+func testSpendUnconfirmed(miner *rpctest.Harness,
+	alice, bob *lnwallet.LightningWallet, t *testing.T) {
+
+	// First we will empty out bob's wallet, sending the entire balance
+	// to alice.
+	bobBalance, err := bob.ConfirmedBalance(0)
+	if err != nil {
+		t.Fatalf("unable to retrieve bob's balance: %v", err)
+	}
+	alicePkScript := newPkScript(t, alice, lnwallet.WitnessPubKey)
+	txFeeRate := chainfee.SatPerKWeight(2500)
+	txFee := btcutil.Amount(28760)
+	output := &wire.TxOut{
+		Value:    int64(bobBalance - txFee),
+		PkScript: alicePkScript,
+	}
+	tx := sendCoins(t, miner, bob, alice, output, txFeeRate, true, 1)
+	txHash := tx.TxHash()
+	assertTxInWallet(t, alice, txHash, true)
+	assertTxInWallet(t, bob, txHash, true)
+
+	// Next we will send a transaction to bob but leave it in an
+	// unconfirmed state.
+	bobPkScript := newPkScript(t, bob, lnwallet.WitnessPubKey)
+	output = &wire.TxOut{
+		Value:    btcutil.SatoshiPerBitcoin,
+		PkScript: bobPkScript,
+	}
+	tx = sendCoins(t, miner, alice, bob, output, txFeeRate, false, 1)
+	txHash = tx.TxHash()
+	assertTxInWallet(t, alice, txHash, false)
+	assertTxInWallet(t, bob, txHash, false)
+
+	// Now, try to spend some of the unconfirmed funds from bob's wallet.
+	alicePkScript = newPkScript(t, alice, lnwallet.WitnessPubKey)
+	output = &wire.TxOut{
+		Value:    btcutil.SatoshiPerBitcoin * 0.5,
+		PkScript: alicePkScript,
+	}
+	tx = sendCoins(t, miner, bob, alice, output, txFeeRate, false, 0)
```

Comment:
similarly, before calling this, attempt to send coins using minconfs=1 and make sure that fails.

Reply:
Updated

Reply:
Same here, no change?